src/pages/md-asset-depreciation/form/AssetsForm.js

```diff
@@ -0,0 +1,324 @@
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import * as yup from 'yup'
+import { FixedAssetsRepository } from 'src/repositories/FixedAssetsRepository'
+import { SystemFunction } from 'src/resources/SystemFunction'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { Grid } from '@mui/material'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import FormShell from 'src/components/Shared/FormShell'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+import { useForm } from 'src/hooks/form'
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { useInvalidate } from 'src/hooks/resource'
+
+export default function AssetsForm({ obj, maxAccess: access, labels, window }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const [data, setData] = useState([])
+  const [preiewPressed, setPreviewPressed] = useState(false)
+
+  const { documentType, maxAccess, changeDT } = useDocumentType({
+    functionId: SystemFunction.AssetsDepreciation,
+    access: access,
+    enabled: !obj?.recordId
+  })
+
+  const invalidate = useInvalidate({
+    endpointId: FixedAssetsRepository.AssetsDescription.snapshot
+  })
+
+  const formatDate = date => {
+    const d = date
+    const month = d?.getMonth() + 1
+    const day = d?.getDate()
+    const year = d?.getFullYear()
+
+    return `${month}-${day}-${year}`
+  }
+  async function fetchData() {
+    const response = await getRequest({
+      extension: FixedAssetsRepository.AssetsDescription.preview,
+      parameters: `_asOfDate=${formatDate(formik.values.date)}`
+    })
+    setPreviewPressed(true)
+    setData(response)
+  }
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      recordId: null,
+      reference: '',
+      date: new Date(),
+      notes: '',
+      status: 1,
+      dtId: documentType?.dtId,
+      plantId: ''
+    },
+    validateOnChange: true,
+    validationSchema: yup.object({}),
+    onSubmit: async obj => {
+      const formattedObj = {
+        ...obj,
+        date: formatDateToApi(obj.date)
+      }
+
+      const response = await postRequest({
+        extension: FixedAssetsRepository.AssetsDescription.set2,
+        record: JSON.stringify({ header: formattedObj, items: data.list || [] })
+      })
+
+      if (!obj.recordId) {
+        toast.success(platformLabels.Added)
+
+        formik.setFieldValue('recordId', response.recordId)
```

Comment:
remove  `formik.setFieldValue('recordId', response.recordId)` , recordId will be updated in getData function below